mmcv/runner/hooks/checkpoint.py

```diff
@@ -58,10 +58,10 @@ def after_train_epoch(self, runner):
 
         # save checkpoint for following cases:
         # 1. every ``self.interval`` epochs
-        # 2. reach the last epoch of training
-        if self.every_n_epochs(
-                runner, self.interval) or (self.save_last
-                                           and self.is_last_epoch(runner)):
+        # 2. reach the last `self.save_last` epochs of training
+        if self.every_n_epochs(runner, self.interval) or (
```

Comment:
This would be better.
```python
if self.every_n_epochs(runner, self.interval) \
    or (self.save_last and self.near_last_epoch(runner, self.save_last)):
```